src/os/bluestore/KernelDevice.cc

```diff
@@ -749,7 +763,10 @@ void KernelDevice::_aio_log_finish(
 	   << std::hex << offset << "~" << length << std::dec << dendl;
   if (cct->_conf->bdev_debug_inflight_ios) {
     std::lock_guard l(debug_lock);
-    debug_inflight.erase(offset, length);
+    if (debug_inflight_write.intersects(offset, length))
```

Comment:
you might want to use lower_bound function to locate the range and then use resulting iterator to erase it - saving one lookup this way

Reply:
well, this will need some modifications in interval_set to implement partial erase for iterator... Perhaps an overkill for this secondary debugging feature...